plugin/arcconf/arcconf.py

```diff
@@ -0,0 +1,919 @@
+            decoded_json = json.loads(cntrl_info_json)
+            getconfig_cntrls_info.append(decoded_json)
+
+        for cntrl in range(total_cntrl):
+            ctrl_name = getconfig_cntrls_info[cntrl]['Controller']['deviceVendor']
+            ctrl_name = ctrl_name + ' ' + getconfig_cntrls_info[cntrl]['Controller']['deviceName']
+            sys_id = str(cntrl + 1)
+            status = getconfig_cntrls_info[cntrl]['Controller']['controllerStatus']
+            if status == CONTROLLER_STATUS_WORKING:
+                status = System.STATUS_OK 
+            status_info = '"Controller Status"=[%s]' % (status)
+            fw_ver = getconfig_cntrls_info[cntrl]['Controller']['firmwareVersion']
+            plugin_data = getconfig_cntrls_info[cntrl]['Controller']['physicalSlot']
+            read_cache_pct = System.READ_CACHE_PCT_UNKNOWN
+            
+            hwraid_mode = getconfig_cntrls_info[cntrl]['Controller']['functionalMode']
+            if hwraid_mode == HW_RAID_EXPOSE_RAW_DEVICES or hwraid_mode == HW_RAID_HIDE_RAW_DEVICES:
+                mode = System.MODE_HARDWARE_RAID
+            elif hwraid_mode == HBA_MODE:
+                mode = System.MODE_HBA
+            else:
+                mode = System.MODE_UNKNOWN
+            status_info += ' mode=[%s]' % str(hwraid_mode)
+            
+            rc_lsm_syss.append(System(sys_id, ctrl_name, status, status_info, plugin_data, _fw_version=fw_ver, _mode=mode, _read_cache_pct=read_cache_pct))
+
+        return rc_lsm_syss
+
+ 
+    @staticmethod
+    def _arcconf_array_to_lsm_pool(arcconf_array, ld_name, sys_id, ctrl_num):
+        pool_id = '%s:%s' %(ctrl_num,ld_name)
+        name = 'Array' + str(ld_name)
+        elem_type = Pool.ELEMENT_TYPE_VOLUME | Pool.ELEMENT_TYPE_VOLUME_FULL
+        unsupported_actions = 0
+        free_space = 0
+        total_space = 0
+        #TODO Need to have a logic for finding free space and total space in the pool.      
+
+        status = Pool.STATUS_OK
+        status_info = ''
+        plugin_data = "%s:%s" % (ctrl_num, ld_name)
+
+        return Pool(
+            pool_id, name, elem_type, unsupported_actions,
+            total_space, free_space, status, status_info,
+            sys_id, plugin_data)
+
+    @_handle_errors
+    def pools(self, search_key=None, search_value=None,
+              flags=Client.FLAG_RSVD):
+        """
+        """
+        lsm_pools = []
+        controllerInfo = {'Status':'Optimal', 'Slot':'255', 'Mode':'HBA', 'Name':'Adaptec', 'Serial Number':'1234'}
+        getconfig_disks_info = []
+        total_cntrl = 0
+        test = self._arcconf_exec(['list' ], COMPONENT[5], False)
+        output_lines = [
+        l for l in test.split("\n")]
+
+        sys_id = ''
+        for line_num in range(len(output_lines)):
+            cur_line = output_lines[line_num]
+            tmp = cur_line.split(":")
+            if tmp[0].strip() == 'Controllers found':
+                total_cntrl = int(tmp[1].strip())
```

Comment:
Why we do output parse here when we already have json parser and `_parse_arcconf_output()`?

And the code like `controllerInfo['Slot'] = cntrl_info[1].strip()` seems not a good practice, try use hash table in parser.